system/free_tcp_port.go

```diff
@@ -1,18 +1,33 @@
 package system
 
 import (
-	"net"
-	"strconv"
+        "net"
+        "strconv"
+        "errors"
 )
 
-func FindFreePort() (int, error) {
-	l, _ := net.Listen("tcp", ":0")
-	defer l.Close()
+func FindFreePort(num int) (int, error) {
+        t := strconv.Itoa(num)
+        toto := ":" + t
```

Comment:
consider renaming variable toto into laddr as in the prototype of Net.Listen https://golang.org/pkg/net/#Listen